jgiven-gradle-plugin/src/main/java/com/tngtech/jgiven/gradle/internal/AbstractJGivenReportImpl.java

```diff
@@ -18,13 +18,13 @@ public AbstractJGivenReportImpl( String name, Task task, String relativeEntryPat
     }
 
     @Override public void configure( ReportGenerator generator ) {
-        generator.setTargetDirectory( getDestination() );
-        generator.setFormat( getFormat() );
-        generator.getConfig().setCustomCssFile( getCustomCssFile() );
-        generator.getConfig().setCustomJsFile( getCustomJsFile() );
-        generator.getConfig().setExcludeEmptyScenarios( isExcludeEmptyScenarios() );
-        if (getTitle() != null) {
-            generator.getConfig().setTitle( getTitle() );
+        generator.addFlag( "--targetDir=" + getDestination() );
```

Comment:
The decision to use string constants benefits the extensibility of the ReportGenerator. It only knows the format and calls the respective report based on that, but it should not know the implementation details of that report. 

The ReportGenerator is mainly a command line interface + the added interface to provide it with flags when instantiated as an object which has a single end point and doesn't have to be reimplemented `generate(List<String> args`.

The problem lies in the custom flags that every report may have - how does one access these custom configurations from the ReportGenerator that sees only an AbstractReport? 

A compromise would be to set all already possibly known flag options like `sourceDir/targetDir` via a custom function, but every custom flag would still be passed as string, but that would be a weird interface to use.

I thought that the unified access point should be the command line flags, as this is the main usage of the reports.